test/suites/integration/bsp/bsp-thresholds.test.ts

```diff
@@ -0,0 +1,175 @@
+
+    it("Shouldn't be able to setGlobalParams without sudo", async () => {
+      const { extSuccess, events } = await api.sealBlock(
+        api.tx.fileSystem.setGlobalParameters(13, 37)
+      );
+
+      strictEqual(extSuccess, false, "Extrinsic should be unsuccessful");
+      const { data } = api.assert.eventPresent("system", "ExtrinsicFailed", events);
+      const error = data[0].toString();
+      strictEqual(error, "BadOrigin", "Extrinsic should fail with BadOrigin");
+
+      strictEqual(
+        (await api.query.fileSystem.blockRangeToMaximumThreshold()).toNumber(),
+        1,
+        "Threshold should not have changed"
+      );
+      strictEqual(
+        (await api.query.fileSystem.replicationTarget()).toNumber(),
+        1,
+        "Replication Target should not have changed"
+      );
+    });
+
+    it("Reputation increased on successful storage", { skip: "Not Implemented" }, async () => {
+      const repBefore = (await api.query.providers.backupStorageProviders(ShConsts.DUMMY_BSP_ID))
+        .unwrap()
+        .reputationWeight.toBigInt();
+      await api.file.newStorageRequest("res/cloud.jpg", "test/cloud.jpg", "bucket-1");
+      await api.wait.bspVolunteer();
+      await api.wait.bspStored();
+
+      const repAfter = (await api.query.providers.backupStorageProviders(ShConsts.DUMMY_BSP_ID))
+        .unwrap()
+        .reputationWeight.toBigInt();
+
+      assert(
+        repAfter > repBefore,
+        "Reputation should increase after successful storage request fufilled"
+      );
+      console.log(`Reputation increased from ${repBefore} to ${repAfter}`);
+    });
+
+    it("zero reputation can still volunteer and be accepted", async () => {
+      // Create a new BSP and onboard with no reputation
+      await addBsp(api, bspDownKey, {
+        name: "sh-bsp-down",
+        bspKeySeed: bspDownSeed,
+        bspId: ShConsts.BSP_DOWN_ID,
+        additionalArgs: ["--keystore-path=/keystore/bsp-down"],
+        bspStartingWeight: 0n
+      });
+      await api.sealBlock(api.tx.sudo.sudo(api.tx.fileSystem.setGlobalParameters(5, 1)));
+
+      await api.file.newStorageRequest("res/smile.jpg", "test/smile.jpg", "bucket-1"); // T0
+      await api.wait.bspVolunteer();
+
+      const events = await api.query.system.events();
+      const matchedEvents = api.assert.eventMany("fileSystem", "AcceptedBspVolunteer", events); // T1
+
+      assert(matchedEvents.length === 2, "Multiple BSPs should be able to volunteer");
+
+      const filtered = matchedEvents.filter(
+        ({ event }) =>
+          (api.events.fileSystem.AcceptedBspVolunteer.is(event) && event.data.bspId.toString()) ===
+          ShConsts.BSP_DOWN_ID
+      );
+
+      assert(
+        filtered.length === 1,
+        "Zero reputation BSP should be able to volunteer and be accepted"
+      );
+      await api.docker.stopBspContainer("sh-bsp-down");
+    });
+
+    // Not sure if this is good test? the times are very dependent on threshold created
```

Comment:
I think it is a good test, but you can make it less "manual" using the runtime API `query_earliest_file_volunteer_block` which would tell you when a BSP can volunteer for a block.

Reply:
ahh very convenient, thanks!